File: petclinic-java-gradle-react-tailwind-ts/src/main/java/dev/pollito/petclinic_java_gradle_react_tailwind_ts/vet/VetService.java
package dev.pollito.petclinic_java_gradle_react_tailwind_ts.vet;

import dev.pollito.petclinic_java_gradle_react_tailwind_ts.generated.model.VetDTO;
import org.springframework.data.domain.Page;
import org.springframework.data.domain.Pageable;

public interface VetService {

    Page<VetDTO> findAll(String filter, Pageable pageable);

    VetDTO get(Integer id);

    Integer create(VetDTO vetDTO);

    void update(Integer id, VetDTO vetDTO);

    void delete(Integer id);

}

File: petclinic-kt-gradle-thymeleaf-tailwind-webpack-htmx-ts/src/main/kotlin/dev/pollito/petclinic_kt_gradle_thymeleaf_tailwind_webpack_htmx_ts/vet/VetService.kt
package dev.pollito.petclinic_kt_gradle_thymeleaf_tailwind_webpack_htmx_ts.vet

import org.springframework.data.domain.Page
import org.springframework.data.domain.Pageable

interface VetService {

    fun findAll(filter: String?, pageable: Pageable): Page<VetDTO>

    fun `get`(id: Int): VetDTO

    fun create(vetDTO: VetDTO): Int

    fun update(id: Int, vetDTO: VetDTO)

    fun delete(id: Int)
}
